src/app/modules/reports/components/time-entries-table/time-entries-table.component.spec.ts

```diff
@@ -76,6 +79,34 @@ describe('Reports Page', () => {
       expect(component.dtTrigger.next).toHaveBeenCalled();
     });
 
+    it('when the uri starts with http or https it should return true and open the url in a new tab', () => {
+      const url = 'http://customuri.com';
+      spyOn(component, 'isURL').and.returnValue(true);
+      spyOn(window, 'open');
+
+      expect(component.openURLInNewTab(url)).not.toEqual('');
+      expect(window.open).toHaveBeenCalledWith(url, '_blank');
+    });
+
+    it('when the uri starts without http or https it should return false and not navigate or open a new tab', () => {
+      const uriExpected = timeEntry.uri;
+      spyOn(component, 'isURL').and.returnValue(false);
+
+      expect(component.openURLInNewTab(uriExpected)).toEqual('');
+    });
+
+    const params = [
+      {url: 'http://example.com', expected_value: true, with: 'with'},
+      {url: 'https://example.com', expected_value: true, with: 'with'},
+      {url: 'no-url-example', expected_value: false, with: 'without'}
+    ];
+    params.map((param) => {
+      it(`when the url starts ${param.with} http or https it should return ${param.expected_value}`, () => {
```

Comment:
A better test name could be `Given the url ${params.url}, the method isURL should return ${params.expected_value}`. In that way, you get rid of the `with` parameter.